specification/deviceregistry/resource-manager/Microsoft.DeviceRegistry/preview/2023-11-01-preview/asset.json

```diff
@@ -0,0 +1,663 @@
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "properties": {
+            "type": {
+              "type": "string",
+              "description": "The extended location type."
+            },
+            "name": {
+              "type": "string",
+              "description": "The extended location name."
+            }
+          },
+          "description": "The extended location."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "description": "Asset definition."
+    },
+    "AssetPatchPayload": {
+      "type": "object",
+      "properties": {
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-mutability": [
+            "read",
+            "create",
+            "update"
+          ],
+          "description": "Resource tags."
+        },
+        "properties": {
+          "$ref": "#/definitions/AssetProperties",
+          "description": "Asset resource properties."
+        }
+      },
+      "description": "Asset definition for Patch operation."
+    },
+    "AssetProperties": {
+      "description": "Defines the asset properties.",
+      "type": "object",
+      "properties": {
+        "uuid": {
+          "type": "string",
+          "description": "Globally unique, immutable, non-reusable id.",
+          "readOnly": true
+        },
+        "assetType": {
+          "type": "string",
+          "description": "Resource path to asset type (model) definition."
+        },
+        "enabled": {
+          "type": "boolean",
+          "description": "Enabled/Disabled status of the asset."
+        },
+        "externalAssetId": {
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "description": "Asset id provided by the customer."
+        },
+        "displayName": {
+          "type": "string",
+          "description": "Human-readable display name."
+        },
+        "description": {
+          "type": "string",
+          "description": "Human-readable description of the asset."
+        },
+        "assetEndpointProfileUri": {
+          "type": "string",
+          "x-ms-mutability": [
+            "read",
+            "create"
+          ],
+          "description": "A reference to the asset endpoint profile (connection information) used by brokers to connect to an endpoint that provides data points for this asset. Must have the format <ModuleCR.metadata.namespace>/<ModuleCR.metadata.name>."
+        },
+        "version": {
+          "type": "integer",
+          "format": "int32",
+          "description": "An integer that is incremented each time the resource is modified.",
+          "readOnly": true
+        },
+        "manufacturer": {
```

Comment:
Consider combining `manufacturerUri` and `manufacturer` an object with nested properties such as `name` and `uri`. In the future if new manufacturer properties keep coming in it will make it unmaintainable to keep adding prefix "manufacturer" to all properties. For example,

```json
{
 "manufacturer": {
     "name": "Foo",
     "uri": "uri",
     "newProp": "newProp"
}
```

vs 

```json
manufacturer: "Foo",
manufacturerUri: "uri"
manufacturerNewProp: "New prop"
```

Reply:
Manufacturer and manufacturerURI are root properties as we don't expect to have any additional elements for manufacturer going forward. This was also to align with the expected format of the resource on the edge that is consumed by OPCUA broker.